client/src/board/boardwidget.cpp

```diff
@@ -778,6 +768,61 @@ void BoardWidget::rightClicked()
     rotateTile(true);
 }
 
+void BoardWidget::startHoldTimer(const std::shared_ptr<NXButtonRailTile>& nxButton)
+{
+  stopTimerAndReleaseButtons();
+
+  m_releaseButton1 = nxButton;
+
+  m_timer = new QTimer(this);
+  m_timer->setInterval(nxButtonHoldTime);
+  m_timer->setSingleShot(true);
+
+  connect(m_timer, &QTimer::timeout, this, &BoardWidget::stopTimerAndReleaseButtons);
+  m_timer->start();
+}
+
+void BoardWidget::startReleaseTimer(const std::shared_ptr<NXButtonRailTile> &firstButton,
+                                     const std::shared_ptr<NXButtonRailTile> &nxButton)
+{
+  // Do not release first button yet
+  m_releaseButton1.reset();
+
+  stopTimerAndReleaseButtons();
+
+  m_releaseButton1 = firstButton;
+  m_releaseButton2 = nxButton;
+
+  m_timer = new QTimer(this);
```

Comment:
I suggest to add a `assert(m_timer);` here, to check if `m_timer` is `nullptr` as expected.